extensions/ql-vscode/src/view/variant-analysis/VariantAnalysisHeader.tsx

```diff
@@ -42,44 +35,65 @@ const Row = styled.div`
 `;
 
 export const VariantAnalysisHeader = ({
-  queryName,
-  queryFileName,
-  totalRepositoryCount,
-  completedRepositoryCount,
-  queryResult,
-  resultCount,
+  variantAnalysis,
   duration,
   completedAt,
-  variantAnalysisStatus,
   onOpenQueryFileClick,
   onViewQueryTextClick,
   onStopQueryClick,
   onCopyRepositoryListClick,
   onExportResultsClick,
   onViewLogsClick,
 }: VariantAnalysisHeaderProps) => {
+  const totalRepositoryCount = useMemo(() => {
+    return variantAnalysis.scannedRepos?.length ?? 0;
+  }, [variantAnalysis.scannedRepos]);
+  const completedRepositoryCount = useMemo(() => {
+    return variantAnalysis.scannedRepos?.filter(repo => [
+      // All states that indicates the repository has been scanned and cannot
+      // change status anymore.
+      VariantAnalysisRepoStatus.Succeeded, VariantAnalysisRepoStatus.Failed,
+      VariantAnalysisRepoStatus.Canceled, VariantAnalysisRepoStatus.TimedOut,
+    ].includes(repo.analysisStatus))?.length ?? 0;
+  }, [variantAnalysis.scannedRepos]);
+  const resultCount = useMemo(() => {
+    const reposWithResultCounts = variantAnalysis.scannedRepos?.filter(repo => repo.resultCount !== undefined);
+    if (reposWithResultCounts === undefined || reposWithResultCounts.length === 0) {
```

Comment:
How about this instead? 

```suggestion
    if (reposWithResultCounts) {
```

I'm happy either way.

Reply:
Oh d'oh! Sorry not sure where my brain went for a minute 😂